tests/misc/tests/misc.js

```diff
@@ -640,4 +641,109 @@ describe("misc", () => {
     assert.ok(account.owner.equals(program.provider.wallet.publicKey));
     assert.ok(account.mint.equals(mint.publicKey));
   });
+
+  it("Can fetch all accounts of a given type", async () => {
+    // Initialize the accounts
+    const data1 = anchor.web3.Keypair.generate();
+    const data2 = anchor.web3.Keypair.generate();
+    const data3 = anchor.web3.Keypair.generate();
+    const data4 = anchor.web3.Keypair.generate();
+    // Initialize filterable data
+    const filterable1 = anchor.web3.Keypair.generate().publicKey;
+    const filterable2 = anchor.web3.Keypair.generate().publicKey;
+    // Set up a secondary wallet and program
+    const anotherProgram = new anchor.Program(
```

Comment:
Here's how I created the secondary program just so I could sign and send it with another wallet. [This relates to this comment](https://github.com/project-serum/anchor/pull/788/files#r715483496)